packages/sdk/src/provider/extensionProviderHelpers/handleUuid.ts

```diff
@@ -0,0 +1,30 @@
+import { v4 as uuidv4 } from 'uuid';
+import { base64Encode } from '../../utils/base64';
+
+interface DappIdentifier {
+  url: string;
+  name: string;
+}
+
+/**
+ * Gets or creates a unique identifier (UUID) based on the provided url and name.
+ * The identifier is stored in localStorage using a Base64 encoded combination of `url` and `name`.
+ *
+ * @param {DappIdentifier} identifier - An object containing the `url` and `name` of the dapp.
+ * @returns {string} - The unique identifier (UUID) for the dapp.
+ */
+function getOrCreateUuidForIdentifier({ url, name }: DappIdentifier): string {
+  const rawIdentifier = url + name;
+  const encodedIdentifier = base64Encode(rawIdentifier);
+
+  let storedUuid = localStorage.getItem(encodedIdentifier) ?? '';
```

Comment:
I think you should check on node and react native to make sure it doesnt make it fail. Otherwise you would have to use the storageManager to be crossplatform (it may fail just because code gets included without being executed in node / rn)

Reply:
this should avoid any fail in case of `localStorage` not available